trieste/models/interfaces.py

```diff
@@ -120,6 +120,13 @@ def optimize(self, dataset: Dataset) -> None:
         """
         raise NotImplementedError
 
+    def after_load(self) -> None:
+        """
+        Call this after de-pickling the model, or loading it in another way that bypasses
+        `__init__`.
+        """
```

Comment:
Perhaps say why we need this. At the moment its not clear until you look in models.py